theseus/src/state/profiles.rs

```diff
@@ -271,9 +286,9 @@ impl Profiles {
             .try_for_each_concurrent(None, |(path, profile)| async move {
                 let json = serde_json::to_vec_pretty(&profile)?;
 
-                let json_path =
-                    Path::new(path.to_str().unwrap()).join(PROFILE_JSON_PATH);
-
+                let json_path = Path::new(path.to_str()
+                    .expect("Could not convert path to string."))
```

Comment:
Expect will still crash the app (it's a panic). Create a new error type which accepts a string and throw it instead using `.map_err(...)?`

Reply:
Done